src/backend/distributed/operations/shard_split.c

```diff
@@ -1780,13 +1753,43 @@ CreateDummyShardsForShardGroup(HTAB *mapOfDummyShardToPlacement,
 				splitShardCreationCommandList,
 				shardInterval->shardId);
 
-			/* Create dummy source shard on the specified placement list */
-			CreateObjectOnPlacement(splitShardCreationCommandList, workerPlacementNode);
+			/* Log resource for cleanup in case of failure only.
+			 * Before we log a record, do a best effort check to see if a shard with same name exists.
+			 * This is because, it will cause shard creation to fail and we will end up cleaning the
+			 * old shard. We don't want that.
+			 */
+			bool relationExists = CheckIfRelationWithSameNameExists(shardInterval,
+																	workerPlacementNode);
 
-			/* Add dummy source shard entry created for placement node in map */
-			AddDummyShardEntryInMap(mapOfDummyShardToPlacement,
-									workerPlacementNode->nodeId,
-									shardInterval);
+			if (relationExists)
+			{
+				ereport(ERROR, (errcode(ERRCODE_DUPLICATE_TABLE),
+								errmsg("Relation %s already exists on worker %s:%d.",
```

Comment:
minor: postgres convention is for error messages to start lower case and not have a period at the end

Reply:
Fixed.